client/cypress/integration/formulaire-tests/formulaire.spec.js

```diff
@@ -0,0 +1,92 @@
+/// <reference types="cypress" />
+
+describe('Formulaire tests', () => {
+  
+  beforeEach(() => {
+  
+    cy.intercept('POST','/graphql',req =>{
+      if (req.body.hasOwnProperty('query') && req.body.query.includes('mutation')){
+        req.alias=`CreateMessage`
+        req.reply((res)=>{
+          res.body.data.createMessage="mock:Votre feedback a été envoyé!"
+        })
+  
+      }
+    
+    }).as('CreateMessage')
+   cy.visit('/')
+  
+
+
+  })
+
+
+  it('Application a le bon titre',()=>{
+    cy.title().should('include','FeedZback')
+  })
+  it('Le champs Votre email zenika est invalid quand il est vide et applique pas le pattern email',()=>{
+    cy.get('#email:invalid').should('have.length',1)
+  })
+  it('Le champs Email Zenika de votre collègue est invalid quand il est vide et applique pas le pattern email',()=>{
+    cy.get('#sendEmail:invalid').should('have.length',1)
+  })
+  it('Le champs Points positifs est invalid quand il est vide',()=>{
+    cy.get('#pointsPositifs:invalid').should('have.length',1)
+  })
+  it("Le champs Axes d'améliorations est invalid quand il est vide",()=>{
+    cy.get('#pointsPositifs:invalid').should('have.length',1)
+  })
+  it('Le formulaire est invalid quand les champs sont invalides aussi', () => {
+
+    cy.get('#feedbackForm:invalid').should('have.length',1)
```

Comment:
Les tests e2e sont très couteux en temps, Donc il vaut mieux grouper les tests ensemble, car à chaque fois que tu fais un it, tu réinitialise le navigateur.
https://docs.cypress.io/guides/references/best-practices#Creating-tiny-tests-with-a-single-assertion

Reply:
Oui je comprends et c'est ce que j'ai fait au début, tu m'as dit que c'est mieux si on précise les testes

Reply:
On c'est mal compris.

Dans un test unitaire jest, ce que tu as fait c'est très bien, mais dans un test e2e, c'est pas une bonne pratique.

Moi j'ai fait la remarque sur le matcher  : 
```
cy.get('#email:invalid').should('have.length',1)
```
Mais plutôt faire un matcher plus précis, mais j'ai vu qu'il est pas possible de faire un `.cy.get('#email').should('be.valid')`, donc il faudrait passer par les classes css ajouté par angular : 

```
cy.get('#email:invalid').should('have.class', 'ng-invalid')
```

Reply:
oui je vois. C'est fait